tox.ini

```diff
@@ -15,7 +15,9 @@ commands = flake8 reactive lib tests unit_tests
 
 [testenv:func]
 basepython = python3
-deps = -r{toxinidir}/test-requirements.txt
+deps = 
+  -c {env:TEST_CONSTRAINTS_FILE:https://raw.githubusercontent.com/openstack-charmers/zaza-openstack-tests/master/constraints/constraints-2024.1.txt}
```

Comment:
I think we should use a local file here instead, at the moment constraints-2024.1.txt is pinning for `python-libjuju>=3.1,<3.2`.

Since we are using the `3/stable` channel, we should have a local constraints file with the content:

```
juju>=3.0.0,<4.0.0
```